### ISSUE TITLE
Parsing error: Unexpected token `import`

### ISSUE THREAD

**Issue by seamlik** [eslint/eslint#11189]
* **ESLint Version:** 5.10.0
* **Node Version:** 11.4.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**

Default.

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": "eslint",
  "parserOptions":{
    "ecmaVersion": "2018",
    "sourceType": "module"
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// Workaround for `--experimental-modules`
// See <https://nodejs.org/docs/latest/api/esm.html#esm_import_meta>
const __dirname = Path.dirname(new URL(import.meta.url).pathname)
```

```bash
eslint sample.mjs
```

**What did you expect to happen?**

A decent report.

**What actually happened? Please include the actual, raw output from ESLint.**

```
  3:40  error  Parsing error: Unexpected token import

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**

No.

**Comment by platinumazure** [comment]
Hi @seamlik, thanks for the issue.

I'm not sure I understand the problem. When using the default parser and `sourceType: "module"`, the word `import` becomes a keyword token that can only be used in import declarations.

I quickly searched Google and it seems this might be using the [`import.meta`](https://github.com/tc39/proposal-import-meta) proposal? If that's the case, it's currently in Stage 3 of the evaluation process, and ESLint's default parser only supports Stage 4 language features. So if your intention is to use that syntax, then you need to use a different parser such as babel-eslint.

**Comment by seamlik** [comment]
Thanks for the quick reply.

Correct, `import.meta` is the currently the only way to use `__dirname` inside Node.js if you use `--experimental-modules`, as described in [this ticket](https://github.com/nodejs/node/issues/16844#issuecomment-342245465).

And so this would be a blocker for many users of ES modules in Node.js, I presume. (Or perhaps I'm the very few who is too experimental on things?)

**Comment by kaicataldo** [comment]
@seamlik That's right. Currently, the solution is to use the [babel-eslint](https://github.com/babel/babel-eslint) parser. That being said, there are plans to [change babel-eslint](https://github.com/babel/babel-eslint/pull/711) to expect babel as a peer dependency. The assumption is that anyone using babel-eslint is also using babel, but you've hit a case where that isn't true.

**Comment by seamlik** [comment]
Now with this config:

```javascript
{
  "extends": "standard",
  "parser": "babel-eslint"
}
```

and installing `babel-eslint`, the error was gone. Thanks for the solution!

Though I personally can't consider this ticket is resolved. But I'll leave the decision to you guys.

**Comment by Lonniebiz** [comment]
Common.js is ancient technology. Anyone using it is more fit as a history teacher than a computer programmer. Here's my [issue](https://stackoverflow.com/questions/54337576/atom-linter-eslint-node-mjs-es-modules-fatal-error).

Why not add another env type in eslint, something like NodeEM?
